traitsui/tests/editors/test_table_editor.py

```diff
@@ -280,216 +300,282 @@ def setUp(self):
     def tearDown(self):
         BaseTestMixin.tearDown(self)
 
-    @requires_toolkit([ToolkitName.qt, ToolkitName.wx])
     def test_table_editor(self):
         object_list = ObjectListWithSelection(
             values=[ListItem(value=str(i ** 2)) for i in range(10)]
         )
+        tester = UITester()
+        with tester.create_ui(object_list, dict(view=simple_view)):
+            pass
 
-        with reraise_exceptions(), \
-                create_ui(object_list, dict(view=simple_view)):
-            process_cascade_events()
-
-    @requires_toolkit([ToolkitName.qt, ToolkitName.wx])
     def test_filtered_table_editor(self):
         object_list = ObjectListWithSelection(
-            values=[ListItem(value=str(i ** 2)) for i in range(10)]
+            values=[ListItem(other_value=i ** 2) for i in range(10)]
         )
+        tester = UITester()
+        with tester.create_ui(object_list, dict(view=filtered_view)) as ui:
+            values = tester.find_by_name(ui, "values")
+            filter = values._target.filter
+            num_filtered_indices = len(values._target.filtered_indices)
+            self.assertIsNotNone(filter)
+            self.assertEqual(num_filtered_indices, 7)
 
-        with reraise_exceptions(), \
-                create_ui(object_list, dict(view=filtered_view)) as ui:
-            process_cascade_events()
-
-            filter = ui.get_editors("values")[0].filter
-
-            process_cascade_events()
-
-        self.assertIsNotNone(filter)
-
-    @requires_toolkit([ToolkitName.qt, ToolkitName.wx])
     def test_table_editor_select_row(self):
         object_list = ObjectListWithSelection(
             values=[ListItem(value=str(i ** 2)) for i in range(10)]
         )
-        object_list.selected = object_list.values[5]
-
-        with reraise_exceptions(), \
-                create_ui(object_list, dict(view=select_row_view)) as ui:
-            editor = ui.get_editors("values")[0]
-            process_cascade_events()
-            if is_qt():
-                selected = editor.selected
-            elif is_wx():
-                selected = editor.selected_row
 
-            process_cascade_events()
+        tester = UITester()
+        with tester.create_ui(object_list, dict(view=select_row_view)) as ui:
+            # click the first cell in the 6th row to select the row
+            values_table = tester.find_by_name(ui, "values")
+            row6_cell = values_table.locate(Cell(5, 0))
+            row6_cell.perform(MouseClick())
+            selected = values_table.inspect(Selected())
 
         self.assertIs(selected, object_list.values[5])
+        self.assertIs(object_list.selected, selected)
 
-    @requires_toolkit([ToolkitName.qt, ToolkitName.wx])
     def test_table_editor_select_rows(self):
         object_list = ObjectListWithSelection(
             values=[ListItem(value=str(i ** 2)) for i in range(10)]
         )
         object_list.selections = object_list.values[5:7]
```

Comment:
would be nice to have shift click functionality enabled to do this type of thing...